web-bundle/src/main/java/com/graphhopper/resources/MapboxResource.java

```diff
@@ -0,0 +1,185 @@
+                throwIllegalArgumentException("Voice units only support metric right now");
+            if (!voiceInstructions)
+                throwIllegalArgumentException("You need to enable voice instructions right now");
+            if (!bannerInstructions)
+                throwIllegalArgumentException("You need to enable banner instructions right now");
+
+            double minPathPrecision = 1;
+            if (overview.equals("full"))
+                minPathPrecision = 0;
+
+            String vehicleStr = convertProfileToGraphHopperVehicleString(profile);
+            List<GHPoint> requestPoints = getPointsFromRequest(pathSegment);
+
+            StopWatch sw = new StopWatch().start();
+
+            // TODO: initialization with heading/bearings
+            // TODO: how should we use the "continue_straight" parameter? This is analog to pass_through, would require disabling CH
+
+            GHRequest request = new GHRequest(requestPoints);
+            request.setVehicle(vehicleStr).
+                    setWeighting(weighting).
+                    setLocale(localeStr).
+                    getHints().
+                    put(CALC_POINTS, true).
+                    put(INSTRUCTIONS, enableInstructions).
+                    put(WAY_POINT_MAX_DISTANCE, minPathPrecision);
+
+            GHResponse ghResponse = graphHopper.route(request);
+
+            float took = sw.stop().getSeconds();
+            String infoStr = httpReq.getRemoteAddr() + " " + httpReq.getLocale() + " " + httpReq.getHeader("User-Agent");
+            String logStr = httpReq.getQueryString() + " " + infoStr + " " + requestPoints + ", took:"
+                    + took + ", " + weighting + ", " + vehicleStr;
+
+            if (ghResponse.hasErrors()) {
+                logger.error(logStr + ", errors:" + ghResponse.getErrors());
+                // Mapbox specifies 422 return type for input errors
+                return Response.status(422).entity(MapboxResponseConverter.convertFromGHResponseError(ghResponse)).
+                        header("X-GH-Took", "" + Math.round(took * 1000)).
+                        build();
+            } else {
+                return Response.ok(MapboxResponseConverter.convertFromGHResponse(ghResponse, request.getLocale())).
+                        header("X-GH-Took", "" + Math.round(took * 1000)).
+                        build();
+            }
+
+        } catch (Exception e) {
+            logger.error("Exception while processing: ", e);
+            throw new MultiException(e);
+        }
+    }
+
+    private void throwIllegalArgumentException(String exceptionMessage) {
+        logger.error(exceptionMessage);
+        throw new IllegalArgumentException(exceptionMessage);
+    }
+
+    private List<GHPoint> getPointsFromRequest(PathSegment pathSegment) {
+
+        // TODO: Could the map at some point break the order?
+        Set<String> pointParams = pathSegment.getMatrixParameters().keySet();
```

Comment:
This seems to be indeed a bit risky. What is the parameter form? Isn't it possible to use the same approach like in RouteResource? `@QueryParam("point") List<GHPoint> requestPoints`

Reply:
The problem is that the points in the Mapbox SDK are so called "MatrixParameters" and not as QueryParameters.

An example would look like this: `...directions/v5/mapbox/driving-traffic/9.453778,48.70906;9.450624,48.706879`

I did some research, but only found this option to read this parameter form. As noted in the TODO, I am not especially happy with this either :).

Reply:
And did you try it for multiple points and does it work?

Reply:
I double checked. It does not work, it seems that it can change the order.

Reply:
I opened an SO post about this here: https://stackoverflow.com/questions/51420380/read-a-list-of-parameters-in-order-12345

Reply:
Ok, according to SO, this is not easily fixable with JAX-RS. I switched to parsing the URL string manually. This seems to work well enough :).